packages/jest-circus/src/utils.js

```diff
@@ -110,46 +110,59 @@ const getEachHooksForTest = (
   return result;
 };
 
+const _makeTimeoutMessage = (timeout, isHook) => {
+  const message = `Exceeded timeout of ${timeout}ms for a ${isHook ? 'hook' : 'test'}.`;
+  return new Error(message);
+};
+
 const callAsyncFn = (
   fn: AsyncFn,
   testContext: ?TestContext,
-  {isHook}: {isHook?: boolean} = {isHook: false},
+  {
+    isHook,
+    timeout,
+    test,
```

Comment:
Shouldn't it be sorted alphabetically? It seems pretty important to @cpojer though, so I'm just stealing his thunder.